### ISSUE TITLE
How to add USB Type C to your USB 2.0 projects – idyl.io

### ISSUE THREAD

**Issue by occoder** [occoder/common_sql#651]
> If you want to replace the micro-USB / mini-USB for a USB type C connectors in your projects, look no further! In this post you will see just how simple it is!

![](https://idyl.io/wp-content/uploads/2020/04/how-to-add-ucb-c-custom-project.jpg)

A USB type C connector connected to a typical FTDI chip

## The USB Type C connector

The USB Type C connector has 24 pins, and is symmetrical so that the cables can be reversed. In effect USB type C is 12 pins. It uses the traditional USB 2.0 D+/D- signals, and two other differential pairs to reach the speeds commanded by the USB 3.2 standard.

![](https://idyl.io/wp-content/uploads/2020/04/USB_Type-C_Receptacle_Pinout.png)

USB Type C receptacle pinout

In addition, it has a bunch of control lines to communicate on type of capability and power delivery.

All in all, it is fairly complex. That being said, if you just want to use a USB Type C as a traditional USB 2.0 device, it is fairly easy to do so.

## USB Type C connector for USB 2.0 applications

If you would like to use a type C plug to replace your micro-USB/mini-USB plugs, all you need to do is:

-   Use a 5.1k pull down resistors on the CC1 line
-   Use a 5.1k pull down resistors on the CC2 line
-   Use the four traditional USB power and data lines: VBUS, GND, D+ and D-
-   Ignore and leave everything else unconnected.

That’s it! It is not more complicated than this!

## The easy connector: The 12 pin type C

There are hundreds of USB type C connectors. Most of the full 24 pin connectors are really unfriendly packages. They are very dense making it incredibly hard to reliably solder. The solution to this is to use a reduced pin count type C. Since most of the lines are left unconnected, 24 pins are not needed.

One of the simplest connectors is 12 pins. A good part number is “TYPE-C-31-M-12” manufactured by [Korean Hroparts Elec](http://www.krhro.com/en/).

![](https://idyl.io/wp-content/uploads/2020/04/20180914_Korean-Hroparts-Elec-TYPE-C-31-M-12_C165948_front.jpg)

A 12 pin USB Type C connectors. Credits: lcsc.com

The connector has the following pinout:

![](https://idyl.io/wp-content/uploads/2020/04/UBS-TYPE-C-31-M-12-footprint-and-pinout.png)

TYPE-C-31-M-12 footprint and pinout

And this is what it looks like on an actual PCB:

![](https://idyl.io/wp-content/uploads/2020/04/USB-c-footprint-on-pcb.jpg)

USB Type C footprint on a PCB

## Example: Connecting to a FTDI chip

The schematics below uses a USB-C 12-pin connector, together with a popular FTDI full speed USB to UART chip (FT231XS)

![](https://idyl.io/wp-content/uploads/2020/04/typical-usb-c-implementation.png)

Typical USB type C connected to a FT231XS

The implementation above is simply the recommended application for a USB 2.0 connector. The only difference? Two 5.1k pull down resistors on the CC1 and CC2 lines of the USB type C receptacle connector.

## In conclusion

There is no more good reason to use micro-USB. Replacing it with USB type C costs two resistors, and a little bit more space on a PCB. These trade-offs are totally worth it as the 2.0 original connectivity is slowly fading away. Soon it’ll be a relic of the past akin to serial cables. You’d better move on now! 
 [https://idyl.io/how-to-add-usb-type-c-micro-usb-replacement/](https://idyl.io/how-to-add-usb-type-c-micro-usb-replacement/)
